Engine/src/Engine/MainMenu.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+#include "UserInterface.h"
+
+class MainMenu : public UserInterface
+{
+
+public:
+
```

Comment:
shouldn't we be defining a destructor here? Since the base class has a pure virtual destructor, I would expect to see an override here